openmdao/vectors/default_transfer.py

```diff
@@ -94,7 +95,15 @@ def _setup_transfers(group):
                     if src_indices is None:
                         pass
                     elif src_indices.ndim == 1:
-                        src_indices = convert_neg(src_indices, meta_out['global_size'])
+                        if isinstance(src_indices, tuple) or \
+                                     (isinstance(src_indices, np.ndarray) and
+                                      src_indices.dtype == object):
+                            if _is_slice(src_indices):
+                                indices = _slice_indices(src_indices, meta_out['size'],
```

Comment:
Use meta_out['global_size'] and meta_out['global_shape'] here.  Note that when a variable is NOT distributed, global_size == size and global_shape == shape.